### PR TITLE
[5] Tests configuration

### LINKED ISSUE #5: Tests configuration
We would like to have integration tests

- Vitest (example https://github.com/vercel/next.js/tree/canary/examples/with-vitest)
- [React testing library](https://testing-library.com/docs/react-testing-library/intro/)

### PR DESCRIPTION
**Pull request by KrisGum**
Closes #5

### CODE REVIEW

File: vitest.config.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import { defineConfig } from 'vitest/config'
```
Comment (KlaraGajaszek) [inline vitest.config.ts:1]: Please add:
- plugin for React: `@vitejs/plugin-react` -  the tasks of this plugin are described on the page https://github.com/vitejs/vite/tree/main/packages/plugin-react#readme
- vite-tsconfig-paths - all information you can fin on this page https://github.com/aleclarson/vite-tsconfig-paths

then add plugins: [react(), tsconfigPaths()], to defineConfig

File: vitest.config.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+
+export default defineConfig({
+    test: {
```
Comment (KlaraGajaszek) [inline vitest.config.ts:4]: please add basic configuration:
`  test: {
    globals: true,
    environment: "jsdom",
     setupFiles: './vitest.setup.ts',
  } `
  
and then in `./vitest.setup.ts'` add 
import '@testing-library/jest-dom/extend-expect';

File: package.json [hunk 3]
```diff
@@ -15,11 +17,13 @@
     "react-dom": "18.2.0"
   },
   "devDependencies": {
+    "@testing-library/react": "^13.3.0",
     "@types/node": "18.7.4",
```
Comment (KlaraGajaszek) [inline package.json:21]: please add @testing-library/user-event - it is really useful when we would like to test a lot of component's actions

**Review by KlaraGajaszek — CHANGES_REQUESTED** [review]

File: vitest.config.ts (hunk 1 above)
Comment (KrisGum) [reply vitest.config.ts:1]: Done.

File: package.json (hunk 3 above)
Comment (KrisGum) [reply package.json:21]: Done.

File: vitest.config.ts (hunk 2 above)
Comment (KrisGum) [reply vitest.config.ts:4]: Done.

According to Vitest page: https://vitest.dev/config/#globals - we should do the following: "To get TypeScript working with the global APIs, add vitest/globals to the types field in your tsconfig.json"..

**Review by jedrzejratajczak — CHANGES_REQUESTED** [review]
Before submitting changes please check if it's everything works even with a simple test (but don't push the test to the repo).
Please add `@testing-library/jest-dom` as a dev dependency as we import it in `vitest.setup.ts`

**Review by jedrzejratajczak — APPROVED** [review]

**Review by KlaraGajaszek — APPROVED** [review]

**Pull request merged**